test/runtests.jl

```diff
@@ -61,19 +62,29 @@ const GHA = GitHubActions
     @test (@capture_out group(() -> println("!"), "a")) == "::group::a\n!\n::endgroup::\n"
 
     mktemp() do file, io
-        withenv("GITHUB_ENV" => file, map(c -> string(c) => nothing, 'a':'z')...) do
-            set_env("a", "b")
-            @test ENV["a"] == "b"
-            @test read(file, String) == "a<<EOF\nb\nEOF\n"
-            set_env("b", "fooEOFbar")
-            @test read(file, String) == "a<<EOF\nb\nEOF\nb<<EOFEOF\nfooEOFbar\nEOFEOF\n"
-            rm(file)
-            set_env("c", [])
-            @test ENV["c"] == "[]"
-            @test read(file, String) == "c<<EOF\n[]\nEOF\n"
-            set_env("d", nothing)
-            @test ENV["d"] == ""
-            @test read(file, String) == "c<<EOF\n[]\nEOF\nd<<EOF\n\nEOF\n"
+        uuid = UUID("9b1deb4d-3b7d-4bad-9bdd-2b0d7b3dcb6d")
+        delimiter = "ghadelimiter_$uuid"
+        mock((uuid4) => Mock(uuid)) do _
+            withenv("GITHUB_ENV" => file, map(c -> string(c) => nothing, 'a':'z')...) do
+                set_env("a", "b")
+                @test ENV["a"] == "b"
+                @test read(file, String) == "a<<$delimiter\nb\n$delimiter\n"
+
+                if VERSION.minor < 6
+                    mock(atexit) do ae
+                        set_env("b", "foo$(delimiter)bar")
+                        @test called_once_with(ae, GHA.fail)
```

Comment:
I haven't used SimpleMock before, but it looks like `called_once_with` doesn't mean "called once with args X, and possibly other times", but rather "called once AND that one time it had args X": https://github.com/JuliaTesting/SimpleMock.jl/blob/4ee3c0dea7238349be2a1ce157684118cb55e135/src/mock_type.jl#L136-L142

Therefore I bet the CI failure is that `atexit` is being called other times too, with other args.

Reply:
I haven't used it before either, I copied the other example from the testsuite. I interpreted that docstring to refer to the specific mock, in this case `ae`, which has only been called once. But I'm not sure.

@christopher-dG do you happen to know?